flang/test/Semantics/OpenMP/loop-association.f90

```diff
@@ -30,6 +30,13 @@
      c = c - 1
   END DO outer
 
+  ! Accept directives between parallel do and actual loop.
+  !$OMP PARALLEL DO
+  !DIR$ VECTOR ALIGNED
+  DO 20 i=1,N
+     a = a + 0.5
+20   CONTINUE
+
```

Comment:
nit: I think this needs an `!$OMP END PARALLEL DO` to be valid openmp

Reply:
For the worksharing loop directives, the closing `END` directive is optional.  However, I agree with the nit and we should have as a matter of style.

Reply:
I didn't add it, because it wasn't in the other tests that do similar loops. But I agree, it's nicer to have, so I have added it now.